Signal/src/ViewControllers/ConversationView/Cells/OWSContactShareView.m

```diff
@@ -18,21 +18,30 @@
 @interface OWSContactShareView ()
 
 @property (nonatomic) ContactShareViewModel *contactShare;
+@property (nonatomic, weak) id<OWSContactShareViewDelegate> delegate;
+
 @property (nonatomic) BOOL isIncoming;
+@property (nonatomic) OWSContactsManager *contactsManager;
+
+@property (nonatomic, nullable) UIView *buttonView;
 
 @end
 
 #pragma mark -
 
 @implementation OWSContactShareView
 
-- (instancetype)initWithContactShare:(ContactShareViewModel *)contactShare isIncoming:(BOOL)isIncoming
+- (instancetype)initWithContactShare:(ContactShareViewModel *)contactShare
+                          isIncoming:(BOOL)isIncoming
+                            delegate:(id<OWSContactShareViewDelegate>)delegate
 {
     self = [super init];
 
     if (self) {
-        _contactShare = contactShare;
-        _isIncoming = isIncoming;
+        self.delegate = delegate;
```

Comment:
Done. This seems like a style point.  I converted some of these properties to `readonly`.

Reply:
Yeah, I guess it's a style thing - but to be clear, it's not just aesthetics.

In general it's risky to call methods on self before self is initialized - since those setters could in turn rely on state that hasn't been initialized (or a future dev could later modify the getters so that they rely on other state). 

That's why Swift strictly enforces this and why I think it's a good convention to enforce upon ourselves.